src/python/impactx/dashboard/Analyze/plot_PhaseSpaceProjections/phaseSpace.py

```diff
@@ -0,0 +1,120 @@
+"""
+This file is part of ImpactX
+
+Copyright 2024 ImpactX contributors
+Authors: Parthib Roy, Axel Huebl
+License: BSD-3-Clause-LBNL
+"""
+
+from ...trame_setup import setup_server
+
+server, state, ctrl = setup_server()
+
+import base64
+import io
+
+from impactx import Config, ImpactX
+
+from ...Input.distributionParameters.distributionMain import (
+    save_distribution_parameters,
+)
+from ...Input.latticeConfiguration.latticeMain import save_lattice_elements
+from ..plot_PhaseSpaceProjections.phaseSpaceSettings import adjusted_settings_plot
+
+# Call MPI_Init and MPI_Finalize only once:
+if Config.have_mpi:
+    from mpi4py import MPI  # noqa
+
+
+def fig_to_base64(fig):
+    """
+    Puts png in trame-compatible form
+    """
+    buf = io.BytesIO()
+    fig.savefig(buf, format="png")
+    buf.seek(0)
+    return base64.b64encode(buf.read()).decode("utf-8")
+
+
+def run_simulation(save_png=True):
```

Comment:
Why does the `phaseSpace.py` analysis logic have a `run_simulation` function?

Reply:
I was not sure how to retrieve the data for the phase space plot, nor did I find anything in the examples where the simulation and phase space plots were done in different functions, which is why I kept it as one.

Reply:
What I mean is: the simulation control should be in its own file.

Once the sim has evolved, you can return the `sim` object or so and plot from it?

Reply:
Yes, this makes sense. I think the simulation script could be located in a folder called 'run'?

Reply:
`pc` is only valid as long as the `sim` object is alive, so it might be safer to return `sim` and query it for post-processing. then finalize and delete it for new runs.